plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePageSink.java

```diff
@@ -338,6 +353,27 @@ private void closeWriter(int writerIndex)
         partitionUpdates.add(wrappedBuffer(partitionUpdateCodec.toJsonBytes(partitionUpdate)));
     }
 
+    @Override
+    public void closeIdleWriters()
+    {
+        for (int writerIndex = 0; writerIndex < writers.size(); writerIndex++) {
+            HiveWriter writer = writers.get(writerIndex);
+            boolean idleWriter = idleWriters.get(writerIndex);
+            // For transactional tables we don't want to split output files because there is an explicit or implicit bucketing
+            // and file names have no random component (e.g. bucket_00000)
+            if (bucketFunction != null
+                    || isTransactional
+                    || !idleWriter
+                    || writer == null
+                    || writer.getWrittenBytes() <= idleWriterMinFileSize) {
```

Comment:
I think it will not set it to `true` when `writer.getWrittenBytes() > idleWriterMinFileSize`